### PR TITLE
Experimental Cache fill plugin (#7470)

### PR DESCRIPTION
**Pull request by vmamidi**
* Experimental Cache fill plugin

* remove trailing whitespace

* Add documentation

* Remove trailing whitespace

(cherry picked from commit 8ac00199deef6c40073de61661bb131de5e8fd5e)

### CODE REVIEW

File: plugins/experimental/cache_fill/background_fetch.cc [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+#include <cinttypes>
+#include <string_view>
+#include <array>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+#include "ts/ts.h"
+#include "ts/remap.h"
+#include "background_fetch.h"
+typedef std::unordered_map<std::string, bool> OutstandingRequests;
```
Comment (brbzull0) [inline plugins/experimental/cache_fill/background_fetch.cc:41]: Seems it was already declared in the H file.

**Comment by ezelkow1** [comment]
[approve ci]

**Comment by ezelkow1** [comment]
[appoce ci autest]

**Comment by vmamidi** [comment]
[approve ci autest]

**Pull request merged**